src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -2034,6 +2034,386 @@ static bool Set(out int x)
             CreateCompilation(source).VerifyDiagnostics();
         }
 
+        [Fact]
+        public void CondAccess_NullCoalescing_01()
+        {
+            var source = @"
+class C
+{
+    void M1(C c)
+    {
+        int x, y;
+        _ = c?.M(out x, out y) ?? true
+            ? x.ToString() // 1
+            : y.ToString();
```

Comment:
Note that DefiniteAssignment has a behavior where once a diagnostic is reported on a variable, it will not report a diagnostic on that variable again. So in order to verify that e.g. a diagnostic is produced or not in both branches, it's more reliable to just use a different variable in each branch.